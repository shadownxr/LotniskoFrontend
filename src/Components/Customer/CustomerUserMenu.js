import React from 'react';
import Button from '@material-ui/core/button';

export default function CustomerUserMenu(props){
    const handleMenuChoice = (choice) => {
        props.menuChoice(choice);
    }

    return(
        <div>
            <div className="UserMenu">
                <Button style={{color: '#fbfd8a'}} onClick={() => handleMenuChoice("View Flights")}>FLIGHTS</Button>
                <Button style={{color: '#fbfd8a'}}  onClick={() => handleMenuChoice("Customer Reservations")}>MY TICKETS</Button>
                <Button style={{color: '#fbfd8a'}}  onClick={() => handleMenuChoice("Online Briefing")}>CHECK IN</Button>
            </div>
        </div>
    )
}