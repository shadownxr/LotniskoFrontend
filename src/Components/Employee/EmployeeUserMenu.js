import React from 'react';
import Button from '@material-ui/core/button';

export default function EmployeeUserMenu(){
    return(
        <div>
            <div className="UserMenu">EmployeeUserMenu<Button>EmployeeUserMenuTest</Button></div>
        </div>
    )
}