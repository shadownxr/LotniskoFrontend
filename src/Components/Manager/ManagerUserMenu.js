import React from 'react';
import Button from '@material-ui/core/button';

/**
 * Manager menu
 * @param {menuChoice} props 
 */
export default function ManagerUserMenu(props){
    const handleMenuChoice = (choice) => {
        props.menuChoice(choice);
    }

    return(
        <div>
            <div className="UserMenu">
                <Button style={{color: '#fbfd8a'}} onClick={() => handleMenuChoice("View Employees")}>CURRENT EMPLOYEES</Button>
                <Button style={{color: '#fbfd8a'}} onClick={() => handleMenuChoice("View Fired Employees")}>FORMER EMPLOYEES</Button>
            </div>
        </div>
    )
}
