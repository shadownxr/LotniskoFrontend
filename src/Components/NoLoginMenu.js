import React from 'react';
import Button from '@material-ui/core/button';

export default function NoLoginMenu(props){
    const handleMenuChoice = (choice) => {
        props.menuChoice(choice);
    }

    return(
        <div>
            <div className="UserMenu"><Button onClick={() => handleMenuChoice("View Flights")}>PrzeglÄ…daj loty</Button></div>
        </div>
    )
}