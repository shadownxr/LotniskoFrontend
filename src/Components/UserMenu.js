import React, { useEffect, useState } from 'react';
import NoLoginMenu from './NoLoginMenu';
import CustomerUserMenu from './Customer/CustomerUserMenu';
import EmployeeUserMenu from './Employee/EmployeeUserMenu';
import ManagerUserMenu from './Manager/ManagerUserMenu';

export default function UserMenu(props){
    const [userType,setUserType] = useState("Not Logged");

    useEffect(() => {
        setUserType(props.userType);
    },[props.userType])

    const renderMenuSwitch = () => {
        switch(userType){
          case "Not Logged":
            return <NoLoginMenu menuChoice={(choice) => props.menuChoice(choice)}/>;
          case "ROLE_USER":
            return <CustomerUserMenu menuChoice={(choice) => props.menuChoice(choice)}/>;
          case "ROLE_EMPLOYEE":
            return <EmployeeUserMenu menuChoice={(choice) => props.menuChoice(choice)}/>;
          case "ROLE_MANAGER":
            return <ManagerUserMenu menuChoice={(choice) => props.menuChoice(choice)}/>;
          default:
            return <NoLoginMenu menuChoice={(choice) => props.menuChoice(choice)}/>;
        }
      }
      return(
        <div>
          {renderMenuSwitch()}
        </div>
      )
}
