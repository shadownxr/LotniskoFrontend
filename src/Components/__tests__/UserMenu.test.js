import React from 'react';
import { shallow } from 'enzyme';

import UserMenu from '../UserMenu';
import NoLoginMenu from '../NoLoginMenu';
import CustomerUserMenu from '../Customer/CustomerUserMenu';
import EmployeeUserMenu from '../Employee/EmployeeUserMenu';
import ManagerUserMenu from '../Manager/ManagerUserMenu';

describe('UserMenu test',() =>{
    it('Renders when loaded',() =>{
        const wrapper = shallow(<UserMenu />);
        expect(wrapper.find(<NoLoginMenu />)).toMatchSnapshot();
    })
    it('Renders ROLE_USER menu on change',() => {
        const role = "ROLE_USER";
        const wrapper = shallow(<UserMenu userType={role}/>);
        expect(wrapper.find(<CustomerUserMenu />)).toMatchSnapshot();
    })
    it('Renders ROLE_EMPLOYEE menu on change',() => {
        const role = "ROLE_EMPLOYEE";
        const wrapper = shallow(<UserMenu userType={role}/>);
        expect(wrapper.find(<EmployeeUserMenu />)).toMatchSnapshot();
    })
    it('Renders ROLE_MANAGER menu on change',() => {
        const role = "ROLE_MANAGER";
        const wrapper = shallow(<UserMenu userType={role}/>);
        expect(wrapper.find(<ManagerUserMenu />)).toMatchSnapshot();
    })
})