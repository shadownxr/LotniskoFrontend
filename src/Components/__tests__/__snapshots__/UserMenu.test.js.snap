// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserMenu test Renders ROLE_EMPLOYEE menu on change 1`] = `ShallowWrapper {}`;

exports[`UserMenu test Renders ROLE_MANAGER menu on change 1`] = `ShallowWrapper {}`;

exports[`UserMenu test Renders ROLE_USER menu on change 1`] = `ShallowWrapper {}`;

exports[`UserMenu test Renders when loaded 1`] = `ShallowWrapper {}`;
